zebra/zebra_nhg.c

```diff
@@ -307,6 +307,25 @@ static int zebra_nhg_insert_id(struct nhg_hash_entry *nhe)
 	return 0;
 }
 
+struct nhg_hash_entry *zebra_presvd_nhg_lookup(struct nhg_hash_entry *lookup)
+{
+	if (lookup == NULL)
+		return lookup;
+	else
+		return hash_lookup(zrouter.nhgs_presvd, lookup);
+}
+
+static int zebra_presvd_nhg_insert(struct nhg_hash_entry *nhe)
+{
+	if (hash_lookup(zrouter.nhgs_presvd, nhe)) {
+		if (IS_ZEBRA_DEBUG_NHG)
+			zlog_debug("%s: NHG %pNG already exists", __func__, nhe);
+		return -1;
```

Comment:
I don't see anywhere in the code for the return code being checked or am I missing it?